src/board/stroke/livestroke.tsx

```diff
@@ -75,92 +68,79 @@ export class BoardLiveStroke implements Tool {
                 this.type === ToolType.Rectangle ||
                 this.type === ToolType.Select
             ) {
-                // eslint-disable-next-line prefer-destructuring
                 this.x = p[0]
-                // eslint-disable-next-line prefer-destructuring
                 this.y = p[1]
             }
         }
-        return pointsSegments
+        this.pointsSegments = pointsSegments
     }
 
-    flatPoints(): number[] {
-        const pointsSegments = this.pointsSegments ?? []
-        if (pointsSegments.length === 0) {
-            return []
-        }
-        if (pointsSegments.length === 1) {
-            return pointsSegments[0]
+    /**
+     * Convert livestroke into the normal stroke format
+     * @param stageScale scale for adjusting the RDP algorithm
+     * @param pageIndex page index of the pageId
+     */
+    finalize(stageScale: number, pageIndex: number): void {
+        this.flatPoints()
+        this.processPoints(stageScale, pageIndex)
+    }
+
+    flatPoints(): void {
```

Comment:
this function should just return the points array instead of storing it to `this.points`, i.e this.points should not be part of livestroke